src/shared/components/TextArea/TextArea.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { forwardRef, useState } from 'react'
+import { getVariant } from '../InputBase'
+import { HelperText, HelperTextCount, HelperTextsWrapper, StyledTextArea, TextAreaWrapper } from './TextArea.style'
+
+export type TextAreaProps = {
+  placeholder?: string
+  onChange?: (event: React.ChangeEvent<HTMLTextAreaElement>) => void
+  value?: string
+  maxLength?: number
+  className?: string
+  helperText?: string
+  warning?: boolean
+  error?: boolean
+  rows?: number
+}
+const TextAreaComponent: React.ForwardRefRenderFunction<HTMLTextAreaElement, TextAreaProps> = (
+  {
+    onChange,
+    placeholder = 'Something should be here',
+    value,
+    maxLength,
+    className,
+    rows = 10,
+    helperText = '\u00A0',
+    warning,
+    error,
+  },
+  ref
+) => {
+  const [charactersWarning, setCharactersWarning] = useState<'warning' | 'error' | undefined>()
+  const [charactersCount, setCharactersCount] = useState(0)
+
+  const handleOnChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    if (onChange) {
+      onChange(e)
+    }
+
+    if (!maxLength) {
+      return
+    }
+
+    const warningLength = maxLength * 0.8
+    const currentLength = e.target.value.length
+
+    setCharactersCount(0)
```

Comment:
Why do we set it to 0 here?

Reply:
Because If a user writes more than 80% of allowed characters and then erases some of them, the counter will not update. It won't disappear - will be stuck on `maxLength * 0.8`. 
Couldn't find a better solution for this.

Reply:
Ah I see now, we display the count if `charactersCount !== 0`. What if we changed that condition to something like `charactersWarning === 'warning' || charactersWarning === 'error'`? Then here in the handler, we can just always call `setCharactersCount` and apply the warning state. I think it could be better readable this way